03_010/client/src/lib/Router.ts

```diff
@@ -0,0 +1,49 @@
+import type { RoutesStore } from '@src/configs/routes';
+import { setState } from './observer';
+
+type ValueOf<T> = T[keyof T];
+
+export default class Router {
+  private routes: RoutesStore;
+  public pageState: string;
+  public setPage: (args: any) => void;
+
+  constructor(routes: RoutesStore, pageState: string) {
+    this.routes = routes;
+    this.pageState = pageState;
+    this.setPage = setState(pageState);
+    this.init();
+  }
+
+  init() {
+    this.handlePopState();
+    window.addEventListener('popstate', this.handlePopState.bind(this));
+  }
+
+  handlePopState() {
+    let { pathname } = location;
+    pathname = pathname === '/' ? pathname : `/${pathname.split('/')[1]}`;
+
+    let Page;
+    for (const [routePath, component] of Object.entries(this.routes)) {
+      if (routePath === pathname) {
+        Page = component;
+        break;
+      }
+    }
+    if (!Page) Page = this.routes['/error'];
```

Comment:
```suggestion
    const Page = Object.entries(this.routes).find(([ routePath, component ]) => routePath === pathName)?.[1] || this.routes['/error'];
```
find 메소드를 이용해서 이렇게 표현할 수 있겠네요!